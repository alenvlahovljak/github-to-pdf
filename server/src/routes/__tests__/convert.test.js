//require neccesary modules
const fs = require("fs");
const path = require("path");
const app = require("../../app");
const request = require("supertest");

//delete JSON
afterAll(() => {
	const dir = path.join(__dirname, "../../public/storage/profiles/json");
	fs.readdir(dir, (err, files) => {
		if (err) throw err;
		for (let file of files) {
			fs.unlink(path.join(dir, file), (err) => {
				if (err) throw err;
			});
		}
	});
});

test("Should create JSON", async () => {
	const res = await request(app)
		.post("/convert/json")
		.send({
			login: "alenvlahovljak",
			name: "Alen Vlahovljak",
			company: "",
			blog: "https://medium.com/@AlenVlahovljak",
			location: "Bosnia and Herzegovina",
			email: "alen_vlahovljak@hotmail.com",
			bio: "Faculty of Information Technologies",
			node_id: "MDQ6VXNlcjQ1NjM5Mjgy"
		})
		.expect(201);
	expect(res.body).toMatchSnapshot();
});
