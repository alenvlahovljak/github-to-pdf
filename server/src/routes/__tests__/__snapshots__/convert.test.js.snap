// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create JSON 1`] = `
Object {
  "message": "JSON saved!",
}
`;
